iceoryx_binding_c/test/moduletests/test_listener.cpp

```diff
@@ -447,4 +460,88 @@ TIMING_TEST_F(iox_listener_test, NotifyingClientEventWithContextDataWorks, Repea
+    iox_listener_attach_service_discovery_event_with_context_data(&m_sut,
+                                                                  serviceDiscovery,
+                                                                  ServiceDiscoveryEvent_SERVICE_REGISTRY_CHANGED,
+                                                                  &serviceDiscoveryCallbackWithContextData,
+                                                                  &serviceDiscoveryStorage);
+    EXPECT_THAT(iox_listener_size(&m_sut), Eq(1));
+
+    iox_listener_detach_service_discovery_event(
+        &m_sut, serviceDiscovery, ServiceDiscoveryEvent_SERVICE_REGISTRY_CHANGED);
+    EXPECT_THAT(iox_listener_size(&m_sut), Eq(0));
+}
+
+void notifyServiceDiscovery(SubscriberPortData& portData)
+{
+    ConditionNotifier(*portData.m_chunkReceiverData.m_conditionVariableDataPtr, 0).notify();
+}
+
+TIMING_TEST_F(iox_listener_test, NotifyingServiceDiscoveryEventWorks, Repeat(5), [&] {
+    ::testing::Test::RecordProperty("TEST_ID", "538a50bc-60c8-4485-b70e-59d0c53f618b");
+    iox_service_discovery_storage_t serviceDiscoveryStorage;
+    EXPECT_CALL(*runtimeMock, getMiddlewareSubscriber(_, _, _)).WillOnce(Return(&m_subscriberPortData[0]));
+
+    iox_service_discovery_t serviceDiscovery = iox_service_discovery_init(&serviceDiscoveryStorage);
+
+    iox_listener_attach_service_discovery_event(
+        &m_sut, serviceDiscovery, ServiceDiscoveryEvent_SERVICE_REGISTRY_CHANGED, &serviceDiscoveryCallback);
+
+    notifyServiceDiscovery(m_subscriberPortData[0]);
+    std::this_thread::sleep_for(TIMEOUT);
+    TIMING_TEST_EXPECT_TRUE(g_serviceDiscoveryCallbackArgument == serviceDiscovery);
+
+    iox_listener_detach_service_discovery_event(
+        &m_sut, serviceDiscovery, ServiceDiscoveryEvent_SERVICE_REGISTRY_CHANGED);
+});
+
+TIMING_TEST_F(iox_listener_test, NotifyingServiceDiscoveryEventWithContextDataWorks, Repeat(5), [&] {
+    ::testing::Test::RecordProperty("TEST_ID", "257c27a5-95c6-489d-919f-125471b399e8");
+    iox_service_discovery_storage_t serviceDiscoveryStorage;
+    EXPECT_CALL(*runtimeMock, getMiddlewareSubscriber(_, _, _)).WillOnce(Return(&m_subscriberPortData[0]));
+
+    iox_service_discovery_t serviceDiscovery = iox_service_discovery_init(&serviceDiscoveryStorage);
+
+    iox_listener_attach_service_discovery_event_with_context_data(&m_sut,
+                                                                  serviceDiscovery,
+                                                                  ServiceDiscoveryEvent_SERVICE_REGISTRY_CHANGED,
+                                                                  &serviceDiscoveryCallbackWithContextData,
+                                                                  &serviceDiscoveryStorage);
+
+    notifyServiceDiscovery(m_subscriberPortData[0]);
+    std::this_thread::sleep_for(TIMEOUT);
+    TIMING_TEST_EXPECT_TRUE(g_serviceDiscoveryCallbackArgument == serviceDiscovery);
```

Comment:
Here we check for the global a second time. I could not see that it was reset after the previous test (or better at the beginning of this one). Otherwise we may get false positive tests. Maybe checking `g_contextData` would be useful as well (in fact, checking only this one would probably be enough).

Reply:
It's reset now during the test setup.